### ISSUE TITLE
Colors as lists of RGB tuples not working

### ISSUE THREAD

**Issue by tmontes** [bokeh/bokeh#2615]
I'm a complete beginner and joined the EuroPython2015 bokeh sprint.
While going through the Quickstart I bumped into this while playing around with colors.

The script below is a variation of the "Vectorized colors and sizes" at http://bokeh.pydata.org/en/latest/docs/quickstart.html. It contains:
* The original colors assignment in the example as a list of "#rrggbb" strings: this works fine.
* A variation with colors built as a list of CSS color name strings: this works fine.
* A variation with colors built as a list of RGB tuples: **this fails**.

I used these declarations commenting/uncommenting them out while experimenting and I also have a ``colors = colors[0]`` to have all the circles colored the same: **this works in all the above cases**.

```
import numpy as np
from bokeh.plotting import figure, output_file, show


N = 4000
x = np.random.random(size=N) * 100
y = np.random.random(size=N) * 100
radii = np.random.random(size=N) * 2


# COLORS as in the original example
colors = [
	"#%02x%02x%02x" % (r, g, 150) for r, g in zip(np.floor(50+2*x), np.floor(30+2*y))
]

# COLORS as strings with CSS colours
colors = ["orange", "pink"] * (N/2)

# COLORS as RGB tuples
colors = [(10, 200, 10), (200, 10, 200)] * (N/2)

# Use the same color all around!
# colors = colors[0]


output_file("color_scatter.html")

p = figure(x_range=(0,100), y_range=(0,100))
p.circle(x,y, radius=radii, fill_color=colors, fill_alpha=0.6, line_color=None)

show(p)
```

This seems to be the 3rd problem that Patrik Stas identified here: https://groups.google.com/a/continuum.io/forum/#!topic/bokeh/BVAPlqMyI1Y but it seems no issue was filed here.

I'd love to be able to fix this but, as I mentioned, I am a bokeh newbie and wouldn't even know where to start.
PS: Thanks for a great project! :)

**Comment by birdsarah** [comment]
This also seems to be a problem when you pass a list of tuples in a ColumnDataSource.

**Comment by rothnic** [comment]
Somehow there was two posts on google groups and my comment got missed. I went through this and listed the issues [in this post](https://groups.google.com/a/continuum.io/forum/?utm_medium=email&utm_source=footer#!msg/bokeh/x3PuNDUdRLA/XC2BsPEndoEJ).

**tl;dr Three problems:**
1) some errors in use
2) scatter doesn't take a vector of color specs
3) hex with alpha is not currently implemented on the python side at least (i implemented rgba tuple)

Original Comment:
---
I did some recent work on color alpha, so I took a look. The thing I focused on was adding support for rgba tuples, but it looks like rgba hex wasn't implemented on the python side either.

Some potential things that could be done on the bokeh side:
- improve errors so that there is better feedback inside of _glyph_functions.py, function _match_data_params. If color inputs don't match any of the checks, it ends up throwing an error about no matching column name.
- add rgba hex, which should be rather easy...the tests would likely be the biggest part
- potentially support lists of ColorSpecs, instead of a single ColorSpec, as it appears is currently supported

On your use of bokeh, here are some observations:
- Your examples are using (a, r, g, b), instead of (r, g, b, a), where (0..255, 0..255, 0..255, 0..1). I'm confident this works, as I created the example you see in the docs, with the source being available here.

- On a quick look, I don't see a specific place saying scatter would take multiple color values in that way. There are some glyphs that do, but almost all examples show coloring the entire group of glyphs the same color. You might be able to do this with columndatasource, as seen here: http://bokeh.pydata.org/en/latest/docs/gallery/iris.html. My recommendation though would be to iterate over the things that are of different colors, then plot those separately, where x and y are arrays/lists, but radius/color are singular. It is likely the way you tried is intended to work, but currently the code that handles the color inputs doesn't correctly identify those as such. This definitely works:

```python
for group, color, radius in zip(groups, colors, radii):
   p.scatter(group.x, group.y, radius=radius, color=color)
show(p)
```

Handling all of the potential combinations of inputs can get pretty complex, so I'd defer to someone else for intended functionality. For example, in your example you are using a single line_color, but an array of fill_color. For my version of bokeh, I don't even get white points for your last examples, it errors out on me, which appears to be due to "Black", instead of "black".

```python
p.scatter([1, 2, 3], [1, 2, 3], radius=[0.05, 0.1, 0.2], fill_color=(0, 0, 255, 0.5), line_color="black")
show(p)
```

**Comment by birdsarah** [comment]
Thanks Nick. Just to clarify the poster of this issue is different from the poster from the mailing list. 

Best, Bird

**Comment by rothnic** [comment]
@birdsarah, no problem. Yeah, was just trying to connect the three threads.

**Comment by canavandl** [comment]
Hi @tmontes 

I've been marking this defect as a duplicate of #2622 (which you actually preceded). It's actually an edge-case of our properties system that doesn't have a great solution. There was a warning added to the documentation in #3904 

I'm going to close this as duplicate, but please reopen if you feel it's the issue is unresolved.

Thanks for joining the sprint!
